packages/components/src/Datalist/Datalist.component.js

```diff
@@ -0,0 +1,341 @@
+						focusedSectionIndex: sectionIndex,
+					});
+				}
+			});
+		} else {
+			const index = this.props.titleMap.findIndex(item => item.name === value);
+			this.setState({
+				focusedItemIndex: index,
+			});
+		}
+	}
+
+	/**
+	 * Set a value.
+	 * This new value can be persisted, or not. If not, it enables the ESC key to reset the value
+	 * @param event The change event
+	 * @param value The new value
+	 * @param persist Value will be persisted if true
+	 */
+	updateValue(event, value, persist) {
+		const previousValue = persist ? value : this.state.previousValue;
+
+		this.setState({
+			// setting the filtered value so it needs to be actual value
+			value: typeof value === 'object' ? value.title : value,
+			previousValue: typeof previousValue === 'object' ? previousValue.title : previousValue,
+		});
+		if (persist) {
+			let enumValue = this.props.titleMap.find(item => item.name === value);
+			if (this.props.multiSection) {
+				this.props.titleMap.forEach(group => {
+					const itemObj = group.suggestions.find(item => item.name === value.title);
+					if (itemObj) {
+						enumValue = itemObj;
+					}
+				});
+			}
+			const payload = { value: get(enumValue, 'value', value) };
+			this.props.onChange(event, payload);
+			this.props.onFinish(event, payload);
+		}
+	}
+
+	/**
+	 * Set back the value to the last validated value.
+	 * This remove the suggestions.
+	 */
+	resetValue() {
+		this.resetSuggestions();
+		this.setState({
+			value: this.state.previousValue,
+		});
+	}
+
+	/**
+	 * Building multiSection items or single section items
+	 * return the items list
+	 */
+	buildGroupItems() {
+		if (this.props.multiSection) {
+			return this.props.titleMap.map(group => ({
+				title: group.title,
+				suggestions: group.suggestions.map(item => ({ title: item.name })),
+			}));
+		}
+		return this.props.titleMap.map(item => item.name);
+	}
+
+	/**
+	 * Filter suggestions.
+	 * This sets at least an empty array, which means the suggestion box will always display
+	 * If the array is empty, the suggestion box will display a "No result" message
+	 * @param value The value to base suggestions on
+	 */
+	updateGroups(value, filterOn) {
+		if (this.props.readOnly || this.props.disabled) {
+			return;
+		}
+
+		// building multiSection items or single section items
+		let groups = this.buildGroupItems();
+		if (value) {
+			// filtering
+			const escapedValue = escapeRegexCharacters(value.trim());
+			const regex = new RegExp(escapedValue, 'i');
+			if (this.props.multiSection) {
+				groups = groups
+					.map(group => ({
+						...group,
+						suggestions: filterOn
+							? group.suggestions.filter(item => filterOn && regex.test(item.title))
+							: group.suggestions,
+					}))
+					.filter(group => group.suggestions.length > 0);
+			} else {
+				// only one group so items are inline
+				groups = filterOn ? groups.filter(itemValue => regex.test(itemValue)) : groups;
+			}
+		}
+
+		this.setState({ groups });
```

Comment:
Rename that into `suggestions`.
When I read `if (!this.state.groups)` somewhere else, I thought it was `if there are no groups` so the code was not clear because of that.